doc/source/api.rst

```diff
@@ -95,6 +97,7 @@ HDFStore: PyTables (HDF5)
    :toctree: generated/
 
    read_hdf
+   HDFStore
```

Comment:
This will also document all methods/attributes (and not all of them are documented in the user guide I think). Not familiar with this part of the code, but just wondering to what extent we need to publicize them (by having html docstring pages for them and listed on the HDFStore page) 
(could also use the class_without_autosummary)